server/media/media.go

```diff
@@ -70,14 +70,41 @@ func matchCORSOrigin(allowed []string, origin string) string {
 	}
 
 	if allowed[0] == "*" {
-		return "*"
+		if origin == "" {
+			return "*"
+		}
+		return origin
 	}
 
-	origin = strings.ToLower(origin)
-	for _, val := range allowed {
-		if strings.ToLower(val) == origin {
+	origin = strings.TrimSpace(strings.ToLower(origin))
+	for _, pattern := range allowed {
+		pattern = strings.TrimSpace(strings.ToLower(pattern))
```

Comment:
This is not the right place to normalize allowed patterns. If this is indeed needed, then please move it to the initialization, for example somewhere here https://github.com/tinode/chat/blob/b6b89594094cc871804f71389b511309bcb993d9/server/media/s3/s3.go#L73